File: Automation testing/Selenium-TestNg/pom-without-pageFactory/src/test/java/com/page/LoginPage.java
package com.page;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.testng.annotations.Test;

public class LoginPage extends BasePage{
	
	public LoginPage(WebDriver driver) {
		super(driver);
	}
	
	By userName=By.name("username");
	By password=By.name("password");
	
	By loginText=By.xpath("//h5");
	By loginButton=By.xpath("//button[text()=\" Login \"]");
	
	public void setUserName(String name) {
		driver.findElement(userName).sendKeys(name);
	}
	
	public void setPassword(String pass) {
		driver.findElement(password).sendKeys(pass);
	}
	
	public void clickLogin() {
		driver.findElement(loginButton).click();
	}
	
	public String getLoginText() {
		return driver.findElement(loginText).getText();
	}
	
	public void login(String strUserName,String strPassword) {
		setUserName(strUserName);
		setPassword(strPassword);
		clickLogin();
	}
	
	
}

File: TypeScript/PlayWright/DemoBlazePOM/page/loginPage.ts
import {Page, Locator} from "@playwright/test"

export class LoginPage{

    readonly page : Page
    readonly username : Locator;
    readonly password : Locator;
    readonly loginSubmit : Locator;

    constructor(page : Page){
        this.page=page
        this.username=page.locator("//*[@id='loginusername']")
        this.password=page.locator("//*[@id='loginpassword']")
        this.loginSubmit=page.locator("//*[@id='logInModal']/div/div/div[3]/button[2]")
    }

    async login(username : string, password : string){
        await this.username.fill(username)
        await this.password.fill(password)
        await this.loginSubmit.click()
    }

}